Modules/_threadmodule.c

```diff
@@ -68,6 +119,20 @@ new_thread_handle(thread_module_state* state)
     return self;
 }
 
+static int
+detach_thread(ThreadHandleObject *handle)
+{
+    assert(get_thread_handle_state(handle) == THREAD_HANDLE_RUNNING);
+
+    // This is typically short so no need to release the GIL
+    if (PyThread_detach_thread(handle->handle)) {
+        PyErr_SetString(ThreadError, "Failed detaching thread");
+        return -1;
+    }
+    set_thread_handle_state(handle, THREAD_HANDLE_DETACHED);
```

Comment:
I don't think the `THREAD_HANDLE_DETACHED` state is useful anymore. We only set it during destruction and then the handle is immediately freed.

Reply:
I have mixed feelings about this. I think there's some value in having the state reflect the transitions that the handle has gone through, even if it happens immediately before the object is freed, for debugging purposes. This is pretty hand-wavy, but in the past it has been super helpful to have extra data like this when groveling through core dumps.

Reply:
Ok, it's fine to keep it then